src/electron/connectivity.ts

```diff
@@ -59,35 +61,38 @@ export function lookupIp(hostname: string): Promise<string> {
       DNS_LOOKUP_TIMEOUT_MS, 'DNS lookup');
 }
 
-// Resolves with true iff a TCP connection can be established with the Shadowsocks server.
-//
-// This has the same function as ShadowsocksConnectivity.isServerReachable in
-// cordova-plugin-outline.
-export function isServerReachable(config: cordova.plugins.outline.ServerConfig) {
-  return lookupIp(config.host || '').then((ip) => {
-    return isServerReachableByIp(ip, config.port || 0);
+// Resolves iff a (TCP) connection can be established with the specified destination in the
+// specified timeout, optionally retrying with a delay.
+export function isServerReachable(
```

Comment:
It's my understanding that the OS will retry TCP connections during the timeout, so it wouldn't be necessary for us to implement the retry logic. Could we increase the timeout instead of retrying?

Reply:
Actually, I understand the motivation of retrying connections. Increasing the timeout wouldn't help for ss-local, since the process wouldn't have been listening on the first attempt.

Reply:
The problem is there's no network delay: if we receive an RST then the test will fail, which will happen unless ss-local opens almost instantly. So, I think we need the retry :-/